capacity-mode-evaluator/README.md

```diff
@@ -0,0 +1,118 @@
+
+### Features
+
+- Retrieves DynamoDB table information and autoscaling settings
+- Fetch CloudWatch metrics for DynamoDB tables and simulate Provisioned Capacity usage and autoscaling based on specified utilization targets and minimum read/write units
+- Provides cost optimization recommendations based on specified utilization targets and minimum read/write units
+- Generates recommendation summary file
+
+**Disclaimer:** This tool makes recommendations based on the number of days specified by `DAYS_LOOK_BACK`, which can be up to the last 14 days. Additionally, when simulating the autoscaling behavior, it cannot consider network delays between scaling events until they are applied in reality. Therefore, it is recommended that recommendations be validated as some of them may not be applicable.
+
+Be aware that changing the capacity mode can have an impact on the applications, so be sure to test any changes thoroughly before making them in production.
+
+## Choosing Capacity Mode in DynamoDB
+
+DynamoDB capacity modes determine how your tables will scale to handle varying levels of read and write traffic. It's important to understand the cost implications of each mode to make informed decisions about which mode to use.
+
+There are two capacity modes available in DynamoDB:
+
+- [**Provisioned Mode:**](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadWriteCapacityMode.html#HowItWorks.ProvisionedThroughput.Manual) In this mode, you specify the read and write capacity for your table and DynamoDB reserves that capacity for your use. You pay a predictable hourly rate for the amount of capacity you provision.
+
+- [**On-Demand Mode:**](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadWriteCapacityMode.html#HowItWorks.OnDemand) In this mode, DynamoDB automatically scales the read and write capacity for your table in response to traffic. You pay for only the requests that you make.
+
+### Provisioned Mode
+
+- Provisioned mode is ideal for workloads that have consistent traffic patterns or predictable spikes in traffic.
+- Provisioned mode provides better performance predictability and enables users to fine-tune capacity to their needs.
+- Provisioned mode is also less expensive than on-demand mode for predictable workloads.
+
+ **Note:** Use auto-scaling to ensure that your tables can handle capacity changes automatically without manual intervention.
+
+### On-demand mode
+
+- On-demand mode is ideal for workloads with unpredictable or infrequent traffic patterns.
+- On-demand mode allows users to pay only for the capacity they consume without having to manually adjust capacity.
+- On-demand mode can be more expensive than provisioned mode for predictable workloads.
+
+**Note:** It's important to monitor your DynamoDB usage and make adjustments to the capacity mode as necessary to optimize cost and performance.
+
+## Requirements
+
+- Python 3.8
```

Comment:
Is this a minimum version, or the only version, or is there a range? Python 3.8 EOL is next year.

Reply:
it's compatible with python versions 3.8 - 3.10.